### PR TITLE
Uniform Color Palette

### LINKED ISSUE #662: Uniform color palette
I have noticed that when coloring embeds for example, [everyone just uses whatever color he likes most.](https://github.com/Together-Java/TJ-Bot/search?l=Java&q=Color)

This is not very pretty and consistent so I suggest we introduce a color palette to use for things like embeds.

We should discuss our favorite color palettes and then maybe have the users vote on the best one.

### PR DESCRIPTION
**Pull request by Nxllpointer**
Closes #662 

This introduces a new `Colors` class that currently contains the Catppuccin Macchiato color palette and the individual feature colors consisting of the palettes colors. I also made the features use their colors from the `Colors` class.

### CODE REVIEW

File: application/src/main/java/org/togetherjava/tjbot/commands/Colors.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    public static final Color LOG_TRACE = COLOR_SKY;
+    public static final Color LOG_DEBUG = COLOR_SAPPHIRE;
+    public static final Color LOG_INFO = SUCCESS;
+    public static final Color LOG_WARN = WARNING;
+    public static final Color LOG_ERROR = FAILURE;
+    public static final Color LOG_FATAL = COLOR_RED;
+
+    public static final Color ROLE_SELECT = COLOR_TEAL;
+
+    public static final Color HELP = COLOR_YELLOW;
```
Comment (Taz03) [inline application/src/main/java/org/togetherjava/tjbot/commands/Colors.java:79]: this is not how it was supposed to be, each feature can use a color no need to make a new variable in a util file for each feature

im very against this

**Review by Taz03 — CHANGES_REQUESTED** [review]

File: application/src/main/java/org/togetherjava/tjbot/commands/Colors.java (hunk 1 above)
Comment (Nxllpointer) [reply application/src/main/java/org/togetherjava/tjbot/commands/Colors.java:79]: This way you can easily change the color palette in the future

Comment (Taz03) [reply application/src/main/java/org/togetherjava/tjbot/commands/Colors.java:79]: y would u want to do that?? make a var at top of each feature for the color it is going to use
this is supposed to be a file which contains color pallet and doesn't get updated unless we decide to switch to a new theme

@Zabuzard whats ur opinion on this??

Comment (Tais993) [reply application/src/main/java/org/togetherjava/tjbot/commands/Colors.java:79]: Yea this is too many colours, adding a colour pallette helps against the 5000 different colours, we should limit it to 5 or something

(a kind of) green for success
(a kind of) red for failure
etc

Comment (Taz03) [reply application/src/main/java/org/togetherjava/tjbot/commands/Colors.java:79]: @Tais993 vote for one dark then

Comment (Tais993) [reply application/src/main/java/org/togetherjava/tjbot/commands/Colors.java:79]: If you're talking about the suggestion you made in the Discord, read my message again, I'm quite saying the opposite of that. I want colours, but not 500 colours, instead like 5

Comment (Taz03) [reply application/src/main/java/org/togetherjava/tjbot/commands/Colors.java:79]: one dark have 7 is that ok?

**Review by Zabuzard — CHANGES_REQUESTED** [review]

File: application/src/main/java/org/togetherjava/tjbot/commands/Colors.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    public static final Color COLOR_SUBTEXT0 = new Color(0xa5adcb);
+    public static final Color COLOR_OVERLAY2 = new Color(0x939ab7);
+    public static final Color COLOR_OVERLAY1 = new Color(0x8087a2);
+    public static final Color COLOR_OVERLAY0 = new Color(0x6e738d);
+    public static final Color COLOR_SURFACE2 = new Color(0x5b6078);
+    public static final Color COLOR_SURFACE1 = new Color(0x494d64);
+    public static final Color COLOR_SURFACE0 = new Color(0x363a4f);
+    public static final Color COLOR_BASE = new Color(0x24273a);
+    public static final Color COLOR_MANTLE = new Color(0x1e2030);
+    public static final Color COLOR_CRUST = new Color(0x181926);
```
[the remark this answers is not in the document]
Comment (Nxllpointer) [reply application/src/main/java/org/togetherjava/tjbot/commands/Colors.java:41]: Yeah that makes sense

File: application/src/main/java/org/togetherjava/tjbot/commands/Colors.java (hunk 1 above)
Comment (Tais993) [reply application/src/main/java/org/togetherjava/tjbot/commands/Colors.java:79]: Didn't know you were talking about the issue, I gave my vote there.

**Comment by Nxllpointer** [comment]
So how do we now actually want to implement this? Use mostly red green yellow, have a color for each feature, store all feature colors in a common place?

**Comment by Zabuzard** [comment]
1. a utility class like u already added, but in the util package
2. the whole color palette as private constants
3. expose some of them as themed constants `public static Color INFO = SKY;` (just an example)

**Comment by Tais993** [comment]
> 1. a utility class like u already added, but in the util package
> 
>     2. the whole color palette as private constants
> 
>     3. expose some of them as themed constants `public static Color INFO = SKY;` (just an example)

Disagree with 2, see issue, I don't want all the colours.

**Comment by Zabuzard** [comment]
im not saying that the whole color palette should be available for usage. but it should be put as private constants into the code, so that we have the whole color palette if needed. or maybe at least link it.

the only available/exposed colors should be the ones we actually use. with well named constants.

**Comment by Tais993** [comment]
Sure!

**Comment by Taz03** [comment]
i'd say make all the colors public and let separate class decide what to use, a util class should not be changed so frequently, every time we want to do something we'll need to make a new var in the utility class, probably for the same color too like say reminder embed and question overview embed want to use the same color then we'll have 2 variables in the util class for same color

**Comment by Zabuzard** [comment]
Nah. We are looking for:
```java
public final class Colors {
  private static final Color ROSEWATER = ...
  private static final Color MAUVE = ...
  ...

  public static final Color INFO = ROSEWATER;
  public static final Color ERROR = MAUVE;
  ...
}
```

**Pull request closed without merging**